apis/cpi/v1alpha1/cpiconfig_types.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CPIConfigSpec defines the desired state of CPIConfig
+type CPIConfigSpec struct {
+	VSphereCPI VSphereCPI `json:"vsphereCPI"`
+}
+
+type NSXTRoute struct {
+	// NSX-T T0/T1 logical router path
+	// +kubebuilder:validation:Optional
+	RouterPath string `json:"routerPath"`
+
+	// Cluster CIDR
+	// +kubebuilder:validation:Optional
+	ClusterCidr string `json:"clusterCidr"`
+}
+
+type NSXT struct {
+	// A flag that enables pod routing
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=false
+	PodRoutingEnabled bool `json:"podRoutingEnabled"`
+
+	// Route configuration for NSXT
+	// +kubebuilder:validation:Optional
+	Routes *NSXTRoute `json:"routes"`
+
+	// A secret reference that contains Nsx-T login credentials to access NSX-T
+	// consists of the field username and password
+	// +kubebuilder:validation:Optional
+	NSXTCredentialsRef *v1.SecretReference `json:"nsxtCredentialsRef"`
```

Comment:
Question: Who creates this credential secret?

Reply:
API user can optionally provider this secret. If they do so, it will override the username and password that are derived from the cluster variable.

Reply:
OK, we probably should have it documented somewhere. Otherwise people would be confused about the usage of these fields.

Reply:
agreed